src/cli/requireWebpackConfig.js

```diff
@@ -104,6 +104,10 @@ export default function requireWebpackConfig(webpackConfig, required, env) {
     config = config(env);
   }
 
+  if (!config.mode) {
+    config.mode = mode;
```

Comment:
I think we should overwrite mode in webpack config when mode is supplied via cli options